src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const newDate = [];
+  const dateSet = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateSet[fromFormat[i]] = dateParts[i];
+  }
+
+  if (dateSet.YYYY) {
+    dateSet.YY = dateSet.YYYY.slice(2);
+  }
+
+  if (dateSet.YY < 30) {
+    dateSet.YYYY = '20' + `${dateSet.YY}`;
+  } else {
+    dateSet.YYYY = '19' + `${dateSet.YY}`;
+  }
```

Comment:
The conversion from 'YY' to 'YYYY' should take into account the 'YY' value from the original date parts, not the one potentially truncated from 'YYYY'. This logic should be applied after the date parts have been split, and only if 'YY' is in the 'fromFormat' and 'YYYY' is in the 'toFormat'.